src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1645,16 +1645,15 @@ private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilde
                 mask = Avx2.Shuffle(mask, shuffleConstant);
 
                 Vector128<byte> res = Sse2.Or(mask.GetLower(), mask.GetUpper());
+                ulong extractedBits = Sse2.X64.ConvertToUInt64(res.AsUInt64());
 
-                // Constant that defines indices of characters within an AVX-Register
-                const ulong indicesConstant = 0xFEDCBA9876543210;
-                ulong extractedBits = Bmi2.X64.ParallelBitExtract(indicesConstant, Sse2.X64.ConvertToUInt64(res.AsUInt64()));
-
-                while (true)
+                for (int idx = i; idx < Vector<ushort>.Count; idx++)
```

Comment:
In this context `Vector<ushort>.Count` isn't safe to use, as (on x86) this can be either AVX or SSE.

Reply:
Oh that should have been `Vector256<ushort>.Count`, thanks for spotting!